### PR TITLE
[T3A3][F12-A3]Wang Yu

### PR DESCRIPTION
**Pull request by 4mirai**

### CODE REVIEW

**Comment by WangYu-g** [comment]
work in progress

File: src/seedu/addressbook/commands/CommandResult.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class CommandResult {
 
     /** The feedback message to be shown to the user. Contains a description of the execution result */
-    public final String feedbackToUser;
+    private final String feedbackToUser;
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/CommandResult.java]: this commit doesn't belong to T3A3.
Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -26,7 +29,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] splitAddress = trimmedAddress.split(",");
+        block = new Block(splitAddress[0].trim());
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: why not 1, or 2, or 5? why 0?
i hope you understand where i am coming from?

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -38,22 +45,71 @@ public static boolean isValidAddress(String test) {
...
-        return value.hashCode();
+        return this.toString().hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
 }
+
+class Block {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: missing header comments.
how will the users know what this is supposed to do? how will any other developer know what they can do with this class?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -12,9 +12,12 @@
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
     private boolean isPrivate;
-
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: are you not enforcing the new format with different components? 
if you are, how are you validating the input address?

**Review by okkhoy — COMMENTED** [review]
please note that you need to update the documentation to reflect these changes you made to the code (in program help and user guide).
also, you need to update the test files to test the new behavior of splitting address to components!

**Comment by okkhoy** [comment]
@WangYu-g some comments added, please check
please close the PR once you have noted the suggestions.

**Pull request closed without merging**